app/pages/certification-redirect.tsx

```diff
@@ -0,0 +1,121 @@
+            id
+          }
+          ...defaultLayout_session
+        }
+        certificationUrlByRowId(rowId: $rowId) {
+          id
+          rowId
+          createdAt
+          applicationByApplicationId {
+            id
+            facilityByFacilityId {
+              facilityName
+              organisationByOrganisationId {
+                operatorName
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+
+  render() {
+    const {query, router} = this.props;
+    const {session} = query;
+    const facility =
+      query?.certificationUrlByRowId?.applicationByApplicationId
+        ?.facilityByFacilityId;
+    const facilityName = facility?.facilityName;
+    const organisationName =
+      facility?.organisationByOrganisationId?.operatorName;
+    const applicationId = this.props.router.query.id;
+    const redirectURI = `/certify?applicationId=${applicationId}`;
+
+    const createdAtMoment = query?.certificationUrlByRowId?.createdAt
+      ? moment(query?.certificationUrlByRowId?.createdAt)
+      : null;
+    const currentMoment = moment();
+    // TODO(Dylan): handle expiry of url validation on the back end
+    // Sets an expiry of 7 days for the certification URL
+    const isValid =
+      currentMoment.format('x') < createdAtMoment?.add(7, 'days').format('x');
+    const certificationUrl = query?.certificationUrlByRowId?.rowId && isValid;
+    return (
+      <>
+        <DefaultLayout
+          title="Submission Certification"
+          session={query.session}
+          needsUser={false}
+          needsSession={false}
+        >
+          {certificationUrl ? (
+            <Row style={{marginTop: '60px'}}>
+              <Col md={{offset: 3, span: 6}}>
+                <h3 className="blue">Your certification is requested.</h3>
+                <p>
+                  On behalf of <strong>{organisationName}</strong>, for facility{' '}
+                  <strong>{facilityName} </strong>
+                  please certify the information reported in the application is
+                  correct.
+                </p>
+                {session?.ciipUserBySub ? (
+                  <>
+                    <p>Please continue to the certification page</p>
+                    <Col md={{span: 5, offset: 1}}>
+                      <Button onClick={async () => router.push(redirectURI)}>
+                        Continue
+                      </Button>
+                    </Col>
+                  </>
+                ) : (
+                  <>
+                    <p>
+                      Please log in or register in order to access this page.
+                      <br />
+                      You will be redirected to the certification page after
+                      doing so.
+                    </p>
+
+                    <Col md={{span: 5, offset: 1}}>
+                      <Form
+                        action={`/login?redirectTo=${decodeURI(redirectURI)}`}
```

Comment:
The `decodeURI` doesn't seem to do much. Investigating this to figure it how it should be handled. There might be a bug in the existing login buttons too

Reply:
Yeah so this is wrong (although somehow handled nicely by the browser and/or express) https://github.com/bcgov/cas-ciip-portal/blob/18f1e3e1ebc7d17e3c1e2882d149fd5a215fd77e/app/components/LoginButton.tsx#L13

We shouldn't decode the URI. Can you please remove the use of `decodeURI` in `LoginButton`/

As for your code above, it should be using [`encodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)

Reply:
done!